common/sgx/verifier.c

```diff
@@ -39,8 +39,10 @@ static oe_result_t _on_unregister(oe_attestation_role_t* context)
 
 static void _free_claim(oe_claim_t* claim)
 {
-    oe_free(claim->name);
-    oe_free(claim->value);
+    if (claim->name != NULL)
```

Comment:
The standard free() is specified to be a no-op if NULL is passed, and so should oe_free.
Hence I think this change should not be needed and it should be annotated however free() is annotated so that no warnings are given when NULL is passed.

Reply:
Indeed both `oe_free` and `free` check for null ptr.

@ryanhsu19 can you share the warning generated for these?

Reply:
I included it because previously scan-build popped up some double-free warnings for many files, but I could not reconstruct it so I will remove it.

Reply:
Fixing double free would require setting `claim->name` and `claim->value` to NULL after free. Not checking for NULL.

Reply:
Couple of questions:

1. Why does scanbuild not care that the `claim` variable is not null-checked before dereferencing (`claim-<name`)?

2. Why does scanbuild care that the parameter to oe_free be null checked. It is valid to pass NULL value to functions.

Reply:
1. Scan build only catches bugs that can actually happen in the execution path. So if no code ever passes null to `_free_claim` then scan-build won't warn about it.
2. It doesn't. I don't think this fixes the reported warning.